### ISSUE TITLE
Broken Hills mine memory crash

### ISSUE THREAD

**Issue by gterror** [BGforgeNet/Fallout2_Restoration_Project#63]
[SLOT09.zip](https://github.com/BGforgeNet/Fallout2_Restoration_Project/files/5213252/SLOT09.zip)
[debug.log](https://github.com/BGforgeNet/Fallout2_Restoration_Project/files/5213253/debug.log)

Every time i enter the mine, the damage from the gas, which ticks, crashes the game to desktop. Using the latest rpu version with no alteration to sfall.

**Comment by burner1024** [comment]
[reporting crashes](https://github.com/BGforgeNet/Fallout2_Unofficial_Patch/blob/master/docs/crash.md)

**Comment by gterror** [comment]
Seems to me that there are missing scripts, but dont know how is that possible

**Comment by burner1024** [comment]
> I thought that its not needed to install the original version?

That is correct. It is likely a newly-introduced one.
I can reproduce the crash. For the time being, working around it with the old script should do fine, with no side effects.

**Comment by burner1024** [comment]
I modified the [script](https://github.com/BGforgeNet/Fallout2_Restoration_Project/blob/7fb2d08cacff2f56fcc1879599587df44c033c15/scripts_src/maps/broken2.ssl#L145) to get some debug: [broken2.int.zip](https://github.com/BGforgeNet/Fallout2_Restoration_Project/files/5215649/broken2.int.zip). For some reason, `party_member_list_critters` returns null pointers. I don't think it's supposed to do that.

```pascal
procedure CauseDamage begin
   variable who;
   ndebug("test0");
   foreach who in party_member_list_critters begin
      ndebug("who is " + who);
      ndebug("dude is " + dude_obj);
      ndebug("who name is " + obj_name(who));
      if not wearing_protective_gear(who)
         and critter_kill_type(who) != KILL_TYPE_robot_kills
      then begin
         ndebug("test1");
         apply_radon_damage(who)
         ndebug("test2");
      end
   end
end
```
```log
broken2: test0

broken2: who is 37695152

broken2: dude is 37695152

broken2: who name is Marcus

broken2: test1

broken2: test3
Loading sound file monitor.ACM...succeeded.

broken2: test4

broken2: test2

broken2: who is 0

broken2: dude is 37695152
Script Error: scripts\broken2.int: op_obj_name: obj is NULL
broken2: who name is Marcus

OPCODE ERROR: critter_inven_obj2() - argument #1 is null.
 > Script: scripts\broken2.int, procedure <couldn't find proc>.Script Error: scripts\broken2.int: op_obj_pid: obj is NULL
OPCODE ERROR: critter_inven_obj2() - argument #1 is null.
 > Script: scripts\broken2.int, procedure <couldn't find proc>.Script Error: scripts\broken2.int: op_obj_pid: obj is NULL
OPCODE ERROR: critter_inven_obj2() - argument #1 is null.
 > Script: scripts\broken2.int, procedure <couldn't find proc>.Script Error: scripts\broken2.int: op_obj_pid: obj is NULL
OPCODE ERROR: critter_inven_obj2() - argument #1 is null.
 > Script: scripts\broken2.int, procedure <couldn't find proc>.Script Error: scripts\broken2.int: op_obj_pid: obj is NULL
OPCODE ERROR: critter_inven_obj2() - argument #1 is null.
 > Script: scripts\broken2.int, procedure <couldn't find proc>.Script Error: scripts\broken2.int: op_obj_pid: obj is NULL
```


@FakelsHub @NovaRain any idea why that happens?

**Comment by NovaRain** [comment]
The crash address 0x42D940 is in `critter_kill_count_type_`. If I remove the `critter_kill_type(who) != KILL_TYPE_robot_kills` check, the crash will be gone, but I also notice other party members don't receive any damage.

I tried to recreate the crash like this:
```pas
// The instruction at 0042d940 referenced memory at 00000064
// The memory could not be read from
foreach who in party_member_list_critters begin
   display_msg("Name is " + obj_name(who));
   display_msg("Kill type: " + critter_kill_type(who));
   critter_dmg(who, 2, DMG_normal_dam BWOR DMG_NOANIMATE BWOR DMG_BYPASS_ARMOR);
end
```
Removing the `critter_dmg` line: the name and kill type are displayed for everyone (I got Sulik, Vic, and Cassidy in party).
Removing the `critter_kill_type` line: only the player's name gets displayed and the player receives 2 point of damage.
Leaving only the `critter_dmg` line: only the player receives 2 point of damage.
